File: src/Z.Core/System.String/String.NullIfEmpty.cs
// Description: C# Extension Methods | Enhance the .NET Framework and .NET Core with over 1000 extension methods.
// Website & Documentation: https://csharp-extension.com/
// Issues: https://github.com/zzzprojects/Z.ExtensionMethods/issues
// License (MIT): https://github.com/zzzprojects/Z.ExtensionMethods/blob/master/LICENSE
// More projects: https://zzzprojects.com/
// Copyright © ZZZ Projects Inc. All rights reserved.
public static partial class Extensions
{
    /// <summary>
    ///     A string extension method that return null if the value is empty else the value.
    /// </summary>
    /// <param name="this">The @this to act on.</param>
    /// <returns>null if the value is empty, otherwise the value.</returns>
    public static string NullIfEmpty(this string @this)
    {
        return @this == "" ? null : @this;
    }
}

File: src (VB.NET)/Z.ExtensionMethods.VB/Z.Core/System.String/String.NullIfEmpty.vb

' Copyright (c) 2015 ZZZ Projects. All rights reserved
' Licensed under MIT License (MIT) (https://github.com/zzzprojects/Z.ExtensionMethods)
' Website: http://www.zzzprojects.com/
' Feedback / Feature Requests / Issues : http://zzzprojects.uservoice.com/forums/283927
' All ZZZ Projects products: Entity Framework Extensions / Bulk Operations / Extension Methods /Icon Library

Public Module Extensions_514

	''' <summary>
	'''     A string extension method that return null if the value is empty else the value.
	''' </summary>
	''' <param name="this">The @this to act on.</param>
	''' <returns>null if the value is empty, otherwise the value.</returns>
	<System.Runtime.CompilerServices.Extension> _
	Public Function NullIfEmpty(this As String) As String
		Return If(this = "", Nothing, this)
	End Function
End Module


